src/zenml/steps/utils.py

```diff
@@ -249,6 +281,10 @@ def get_artifact_config_from_annotation_metadata(
             if artifact_config is not None:
                 raise ValueError(error_message)
             artifact_config = metadata_instance
+        elif callable(metadata_instance):
+            if output_name is not None:
+                raise ValueError(error_message)
+            output_name = metadata_instance
```

Comment:
In line `300` below I think might be another issue: What if someone passes a callable and an artifact config without name. Seems like it then sets the callable on the artifact config, but it does not get evaluated (because the validator already ran)